lib/cadet_web/controllers/user_controller.ex

```diff
@@ -2,32 +2,27 @@ defmodule CadetWeb.UserController do
   @moduledoc """
   Provides information about a user.
   """
-  use CadetWeb, :controller
 
+  use CadetWeb, :controller
   use PhoenixSwagger
 
+  def index(conn, _) do
+    user = conn.assigns.current_user
+    # TODO: fetch total xp for this user
+    render(conn, "user_info.json", name: user.name, role: user.role, xp: 0)
```

Comment:
I think it's preferable you just pass the user directly and use the controller's method name in the view.
```elixir
render(conn, "index.json", user: user, xp: 0)
```